components/status_dropdown/status_dropdown.jsx

```diff
@@ -115,6 +193,24 @@ export default class StatusDropdown extends React.PureComponent {
         const setDnd = needsConfirm ? () => this.showStatusChangeConfirmation('dnd') : this.setDnd;
         const setAway = needsConfirm ? () => this.showStatusChangeConfirmation('away') : this.setAway;
         const setOffline = needsConfirm ? () => this.showStatusChangeConfirmation('offline') : this.setOffline;
+        const setCustomTimedDnd = needsConfirm ? () => this.showStatusChangeConfirmation('dnd') : this.setCustomTimedDnd;
+
+        const dndSubMenuItems = this.dndTimes.map((time, index) => {
+            if (index === 5) {
+                return {
+                    id: `dndTime-${time}`,
+                    direction: 'right',
+                    text: time,
+                    action: () => setCustomTimedDnd(),
+                };
+            }
+            return {
+                id: `dndTime-${time}`,
+                direction: 'right',
+                text: time,
+                action: () => setDnd(event, index),
+            };
+        });
```

Comment:
this can be simplified with using a ternary in `action` property:

```suggestion
        const dndSubMenuItems = this.dndTimes.map((time, index) => ({
                id: `dndTime-${time}`,
                direction: 'right',
                text: time,
                action: index === 5 ? () => setCustomTimedDnd() : () => setDnd(event, index),
            });
        });
```